### PR TITLE
[Inductor]: Support updated Triton `AttrsDescriptor`

### PR DESCRIPTION
**Pull request by alexbaden**
The Triton `AttrsDescriptor` object was refactored in https://github.com/triton-lang/triton/pull/4734. These changes add support for the new `AttrsDescriptor` while maintaining backwards compatibility with the existing version. The main changes are different names for the initialized of the descriptor parameters, and a creation via a static method instead of the class constructor. 

Depends on #137458 which removes some unused logic around the old descriptor. Those changes make this PR cleaner, but if for some reason that old logic is still used I can make adjustments. 

Use of the new `AttrsDescriptor` depends on https://github.com/triton-lang/triton/pull/4888

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: torch/_inductor/runtime/hints.py [hunk 1]
```diff
@@ -27,48 +26,55 @@ class TileHint(Enum):
...
+            equal_to_1=None,
+        ):
+            # Prepare the arguments for AttrsDescriptor
+            kwargs = {
+                "tt.divisibility": divisible_by_16,
+                "tt.equal_to": equal_to_1,
+            }
+
+            # Instantiate AttrsDescriptor with the prepared arguments
+            return AttrsDescriptor.from_dict(kwargs)
```
Comment (jansel) [inline torch/_inductor/runtime/hints.py:54]: Let's add some asserts to ensure things don't change over time (or with different backends).  For example:
```
assert res.property_values["tt.divisibility"] == 16
```

If that changes, this code will be incorrect.

**Review by jansel — CHANGES_REQUESTED** [review]

**Comment by alexbaden** [comment]
@jansel I rebased and addressed your change request. Please let me know if this needs anything else.

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**